tests/test_minimal/test_tools/test_expand_paths.py

```diff
@@ -98,3 +100,47 @@ def test_expand_paths(tmpdir):
     for x in string_directory_out:
         assert x in expected
     assert len(string_directory_out) == len(expected)
+
+
+def test_expand_paths_ibl(tmpdir):
+    expander = LocalPathExpander()
+
+    # set up IBL Steinmetz video file structure
+    generate_path_expander_demo_ibl(tmpdir)
+    base_directory = Path(tmpdir)
+
+    # NOTE: empty brackets used b/c for some reason NR_0021/2022-06-30/ subdir is 002/ instead of 001/
+    out = expander.expand_paths(
+        dict(
+            ibl_video_file=dict(
+                base_directory=base_directory,
+                file_path="steinmetzlab/Subjects/{subject_id}/{session_start_time:%Y-%m-%d}/{}/raw_video_data/_iblrig_leftCamera.raw.{session_id}.mp4",
+            ),
+            ibl_video_directory=dict(
+                base_directory=base_directory,
+                folder_path="steinmetzlab/Subjects/{subject_id}/{session_start_time:%Y-%m-%d}/{}/raw_video_data",
+            ),
+        ),
+    )
+
+    # build expected output from file
+    expected_file_path = Path(__file__).parent / "expand_paths_ibl_expected.json"
+    with open(expected_file_path, "r") as f:
+        expected = json.load(f)
+    for entry in expected:
+        if "NWBFile" in entry["metadata"]:  # `datetime` is not JSON serializable, so convert from string
+            if "session_start_time" in entry["metadata"]["NWBFile"]:
```

Comment:
You should be able to handle decoding and encoding of datetimes using the `neuroconv.utils.NWBMetadataEncoder` class instead of having to manually parse it here

Reply:
`neuroconv.utils.NWBMetadataEncoder` doesn't have decoding implemented, so I think I can either:
1) keep the manual datetime parsing code and compare the unmodified `expand_paths` output against the parsed JSON
2) encode the output of `expand_paths` using `NWBMetadataEncoder` (with `json.loads(json.dumps(..., cls=NWBMetaDataEncoder))`), so I can just load the target JSON file with no parsing and compare. Technically, this approach won't differentiate if `expand_paths` returns string-formatted session start times instead of `datetime.datetime` objects but that feels like a pretty negligible concern (?)

Thoughts between the two?

Reply:
Yes, option (2) is what we usually go with